addons/udes_stock/models/stock_move_line.py

```diff
@@ -808,8 +808,8 @@ def _validate_location_dest(self, location=None):
                       (location.name, picking.location_dest_id.name))
                 )
 
-            if picking.picking_type_id.u_drop_location_constraint == 'enforce':
-
+            if picking.picking_type_id.u_drop_location_constraint \
+                    in ['enforce', 'enforce_with_empty']:
```

Comment:
I think in line 825 we shouldn't raise the error if the location is an empty location and constraint is `enforce_with_empty`

Reply:
Forget about the example, that will accept any empty location, was trying to avoid to call get_empty_location(), but I don't think we can.
```
if constraint == enforce_with_empty and not location.quant_ids:
    locations += get_empty_locations()
if locations and location not in locations:
...    
```

Reply:
Yep, you're right, as the system is allowing to use empty locations if the constraint is `enforce_with_empty`, it shall not raise an error. 

I'm not sure about the `not location.quant_ids` clause because 1) the `_validate_location_dest` call can happen before `write` is completed (synchronously by overriding `write` - see above method) and you may not have the quant update by then (we're looking for that, right?) and 2) what we're trying to do here is to make the system more tolerant and allow using empty locations for whatever reason. I may be missing something; what's the reason for such clause?

Reply:
You are right, was a copy paste from the previous example. Will update the example.